src/relay/backend/aot_executor_codegen.cc

```diff
@@ -126,7 +126,14 @@ class AOTOnDemandAllocator : public transform::DeviceAwareExprVisitor {
     for (const auto& param : func_node->params) {
       CreateStorage(param.get());
     }
-    GetStorage(func_node->body);
+    StorageInfo si = GetStorage(func_node->body);
+
+    // If the final expr could not be found it means it was let bound,
```

Comment:
I suspect if it's better to maintain the invariant GetSorage(let-bound var) == GetStorage(let-bound value) as you visit let bindings. Pushing on this to see how clean we can make working in ANF for future reference.

Reply:
I recall trying this but moved away from the idea due to getting invalid outputs, but I agree it would tidy things up a bit.. I intended to have another look the past couple of days but ended up being busy with other things before I go away, I'll take another look when I return :)